src/main/java/org/folio/rest/impl/TitlesAPI.java

```diff
@@ -27,8 +54,17 @@ public void getOrdersStorageTitles(String query, String totalRecords, int offset
 
   @Override
   @Validate
-  public void postOrdersStorageTitles(Title entity, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-    PgUtil.post(TITLES_TABLE, entity, okapiHeaders, vertxContext, OrdersStorageTitles.PostOrdersStorageTitlesResponse.class, asyncResultHandler);
+  public void postOrdersStorageTitles(Title title, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    pgClient.withTrans(conn -> titleService.saveTitle(title, conn)
+        .onComplete(ar -> {
+          if (ar.failed()) {
+            log.error("Title creation failed, title={}", JsonObject.mapFrom(title).encodePrettily(), ar.cause());
+            asyncResultHandler.handle(buildErrorResponse(ar.cause()));
+          } else {
+            log.info("Title creation complete, id={}", title.getId());
+            asyncResultHandler.handle(buildResponseWithLocation(title, getEndpoint(title)));
+          }
+        }));
```

Comment:
should be only single ')' at the end.
Because now response can be send even before transaction completes.
Also there is 4 spaces instead of 2 for .onComplete

Reply:
fixed